src/terminal/kitty.rs

```diff
@@ -232,15 +232,20 @@ impl KittyImages {
 
         let (image_width, image_height) = image.dimensions();
 
-        let image = if image_width > terminal_size.width || image_height > terminal_size.height {
-            image.resize_to_fill(
-                terminal_size.width,
-                terminal_size.height,
-                FilterType::Nearest,
-            )
-        } else {
-            image
-        };
+        let (image, image_width, image_height) =
+            if image_width > terminal_size.width || image_height > terminal_size.height {
+                (
+                    image.resize_to_fill(
+                        terminal_size.width,
+                        terminal_size.height,
+                        FilterType::Nearest,
+                    ),
+                    terminal_size.width,
+                    terminal_size.height,
```

Comment:
I'm sorry for this blatant bug which I think I introduced when refactored, but I don't think this is entirely correct either.

See, `resize_to_fill` preserves the aspect ratio of the image, and it won't necessarily fill the entire width or height of the terminal, so I think instead of taking the terminal size you'd need to take the size of the resized image again.

Reply:
Thanks for pointing on the aspect ratio. You are right, I should give more focus on little things. 😊will fix it.